adr/telemetry.md

```diff
@@ -0,0 +1,60 @@
+# Title [ADR-1]
+
+Collecting structured, event based diagnostics (Telemetry)
+
+## Status
+
+Proposed
+
+## Context
+
+The project uses [tracing](https://docs.rs/tracing/latest/tracing/) to
+collect this information. The tracing crate provides the helpful
+"instrument" macro for creating/entering tracing spans on function
+invoke.
+
+Currently, our use of this macro is undifferentiated and we are seeing
+issues where our telemetry data is affecting performance of the router
+because of volumes and format (JSON) of collected data.
+
+## Decision
+
+This ADR proposes that we change our approach to using #[instrument]
+so that we only trace the exact information which we need. This
+will require a review of existing usage.
+
+Recommendations:
+ - Always specify `skip_all` when using the #[instrument] macro
+
+e.g.:
+```
+#[instrument(skip_all, ...)]
+```
+
+This will have the effect of making the macro "opt-in" rather than the
```

Comment:
this should explain the difference in behaviour, ie what is opted in or out (the serialization of all arguments and return values in a span)

Reply:
Addressed in: https://github.com/apollographql/router/pull/213/commits/8a213ed756921ef0874de7164b952edc40351a31